### ISSUE TITLE
The accuracy of validation_data in fit and evaluate() are not the same, why?

### ISSUE THREAD

**Issue by sjjlr** [keras-team/keras#13165]
I am now using Keras for image classification with 16 classes. I have 100000 images for training and other 1000 images for validate. When training, I use these 1000 images as validate_data parameter in model.fit(), and get the val_acc as 0.8 after 10 epoches. Then this model is stored. Finally I load this model, and use model.evaluate() to test these 1000 images, the acc is just 0.3. I really don't understand what's happing. 
I just want to see the performance of the stored model, so when model.evaluate(), these 1000 images with known labels are used. But why the accuracy in fit() and evaluate() are not the same?

model.fit(x_train, y_train, batch_size=128, epochs=20, verbose=1, validation_data=(x_test, y_test))
model.evaluate(x_test, y_test, batch_size=128)

**Comment by gowtham-kp** [comment]
@sjjlr For faster response, this question can best be posted in Stackoverflow, Keras Slack channel or any support groups of Keras as this platform is only meant for bug/performance, feature related issues. Thanks!
